src/Visualization/Python/PlotMemoryMonitors.py

```diff
@@ -0,0 +1,269 @@
+            local_totals_df[subfile_name] = total_over_nodes(df)
+
+        totals_df = pd.concat([totals_df, local_totals_df])
+
+    # Restrict to x-bounds if there are any
+    for subfile in memory_filenames:
+        if x_bounds is not None:
+            totals_df[subfile] = totals_df[subfile][
+                (totals_df["Time"] >= x_bounds[0])
+                & (totals_df["Time"] <= x_bounds[1])
+            ]
+
+    # Need .dat because all other components have that extension
+    the_rest_str = "The Rest.dat"
+
+    # Given a dictionary of DataFrames, sum all the columns from all the
+    # DataFrames that list totals for each node to get a grand total memory
+    # usage. Then return a DataFrame with the time and this new total
+    def compute_overall_total_and_the_rest(df: pd.DataFrame):
+        total = None
+        for component_name in memory_filenames:
+            if total is not None:
+                total += df[component_name]
+            else:
+                total = df[component_name].copy()
+
+        average_max = total.mean()
+        # Always plot total
+        components_to_plot = []
+
+        # If the max of a component is too small (less than 1% here) just
+        # group them all together in a category called "The Rest". Otherwise
+        # keep the original component
+        the_rest = None
+        for component_name in memory_filenames:
+            component_total = df[component_name]
+            if component_total.max() < 0.01 * average_max:
+                if the_rest is not None:
+                    the_rest += component_total
+                else:
+                    the_rest = component_total.copy()
+            else:
+                components_to_plot.append(component_name)
+
+        if the_rest is not None:
+            components_to_plot.append(the_rest_str)
+
+        return total, the_rest, components_to_plot
+
+    # Now add the total and the rest to the totals DataFrame
+    total_series, the_rest_series, components_to_plot = (
+        compute_overall_total_and_the_rest(totals_df)
+    )
+    totals_df["Total"] = total_series
+    if the_rest_series is not None:
+        totals_df[the_rest_str] = the_rest_series
+
+    # For plotting in MB vs GB
+    divisor = 1.0 if use_mb else 1000.0
+
+    # Start plotting
+    if stylesheet is not None:
+        plt.style.use(stylesheet)
+
+    fig = plt.figure()
+    ax = fig.add_subplot(111)
+
+    ax.plot(
+        totals_df["Time"],
+        totals_df["Total"] / divisor,
+        color="black",
+        label="Total",
+    )
+
+    # Determine plotting order
+    maxes = [totals_df[component].max() for component in components_to_plot]
+    permutation = np.argsort(maxes)[::-1]
+
+    # Plot the individual components
+    for idx in permutation:
+        component = components_to_plot[idx]
+        ax.plot(
+            totals_df["Time"],
+            totals_df[component] / divisor,
+            linewidth=0.2,
```

Comment:
Is the line width needed because the time sampling is so dense? Generally better to leave style up to the stylesheet.

Reply:
I did this so the total line would stand out in comparison to the other lines

Reply:
Ah ok, maybe the color is enough? Up to you.